### ISSUE TITLE
Could you add PWM of possible remote?

### ISSUE THREAD

**Issue by BorgStefan** [EmanuelFeru/hoverboard-firmware-hack-FOC#28]
Hi Emanuel,

I have taken 4 ten inch wheels and mainboard from 2 Hoverboards. Constructed my own skatebord and using Benjaf repository to make and build a software which uses PWM as remote. Hence I have a 4 wheel drive and it does around 20 km/h. 
https://github.com/benjaf/hoverboard-firmware-hack-esk8
I would be really happy if you could add PWM control to your sw. I would like to try it with my skateboard. My skateboard makes 12 km with 4 Hoverboard batteries.
Many thx for your great effort so far! 
![IMG_9044](https://user-images.githubusercontent.com/48124398/75794125-a5f15b80-5d70-11ea-8fdb-d93d26047062.jpg)

BR Stefan

**Comment by pinguinpfleger** [comment]
Hi Stefan,

PPM is already implemented. 
It is serialized PWM, means all channels goes over one wire.
You'll have to convert PWM to PPM using an arduino / esp / teensy ... 
Code you'll find here on github.

PS: interesting eboard :-)

**Comment by BorgStefan** [comment]
Hi,
I am using hoverboard hack made by Niclas Fausth which Benjalf has modified to work with a PWM remote. I managed to flash to mainboards and use 2 receivers to one PWM based remote.
I will try and modify feru sw to include benjalf modifications.
A Video of my first tries with Benjalf sow.

**Comment by pinguinpfleger** [comment]
I see, https://github.com/benjaf/hoverboard-firmware-hack-esk8/commit/aa910a9b9a90580f96629c2652b98f8cfc7fce3b

Thats nice and would be cool when you do this as new VARIANT_PWM

**Comment by EmanuelFeru** [comment]
That would be nice to add, however is up to you to perform the testing since I don't have the HW.
I agree with Kai to add a new VARIANT_PWM to keep all the rest unaffected. If you do the changes make a Pull Request and I will review.

**Comment by BorgStefan** [comment]
Hi,

Well it has been almost 11 years since I made some coding...
I am not using my skateboard for testing. I have a testbed where I try my new sw changes. Benjaf code works and the setup I made with his code is reliable but would like to try new sw.
Br Stefan

4 mars 2020 kl. 21:47 skrev EmanuelFeru <notifications@github.com>:

﻿

That would be nice to add, however is up to you to perform the testing since I don't have the HW.
I agree with Kai to add a new VARIANT_PWM to keep all the rest unaffected. If you do the changes make a Pull Request and I will review.
